### PR TITLE
Fix and remove unused code in KeyboardLayout widget

### PR DESCRIPTION
**Pull request by k4leg**
#2027

### CODE REVIEW

File: libqtile/widget/keyboardlayout.py [hunk 1]
```diff
@@ -22,107 +22,245 @@
...
+        (
+            'variant',
+            None,
+            "Specifies which variant of the keyboard layout should be used to"
+            " determine the components which make up the keyboard"
+            " description.",
+        ),
     ]
+    dependencies = {
+        'setxkbmap': ['setxkbmap', '-version'],
```
Comment (tych0) [inline libqtile/widget/keyboardlayout.py:148]: Can we depend on either xkb-switch or setxkbmap, but not both?

Comment (k4leg) [reply libqtile/widget/keyboardlayout.py:148]: Yes, we can make setxkbmap an optional dependency (then the user will have to configure it via `/etc/X11/xorg.conf.d/00-keyboard.conf`).

**Comment by k4leg** [comment]
Looks like issue #1130. Tests fail with `TOXENV=py39 tox` in qtile/master.

File: libqtile/widget/keyboardlayout.py [hunk 2]
```diff
@@ -22,107 +22,274 @@
...
 
-    def __init__(self, **config):
-        base.InLoopPollText.__init__(self, **config)
+    def __init__(self, **config) -> None:
+        base._TextBox.__init__(self, **config)
         self.add_defaults(KeyboardLayout.defaults)
+        self.raise_for_dependencies()
+        self.use_optional_dependencies_if_available()
+        event_loop = asyncio.get_event_loop()
+        event_loop.create_task(self.loop())
```
Comment (tych0) [inline libqtile/widget/keyboardlayout.py:157]: Is there a reason to switch to this vs. just continuing to use InLoopPollText? Maybe we should just switch InLoopPollText to use this?

Comment (k4leg) [reply libqtile/widget/keyboardlayout.py]: Okay, there is no particular reason to switch to `_TextBox`.

**Comment by tcld** [comment]
For me `xkb-switch` appears to not be part of any official repositories. This means I have to remove the keyboard widget to keep using Qtile or stay on an older commit for now. Is `xkb-switch` expected to become a part of official repositories for arch and other distros?

**Comment by tych0** [comment]
> I have to remove the keyboard widget to keep using Qtile or stay on an older commit for now

Or send a patch to fix the bug via setxkbmap :)

> Is xkb-switch expected to become a part of official repositories for arch and other distros?

No idea. I don't use this area of the code myself, so independent reviews are helpful. I think if you'd like this reverted, open a new issue and tag @k4leg in it and let's see what happens.

**Comment by tcld** [comment]
I honestly don't think I ever understood the issue with the previous implementation, so I have no base at all to fix anything.

I can't tell without going back and figuring out what the issue was, but I hope that either `xkb-switch` is going to eventually become official or - at least - this new implementation brings a ton of value with it. (Does it?)

**Comment by tych0** [comment]
> I honestly don't think I ever understood the issue with the previous implementation, so I have no base at all to fix anything.

I didn't either; there are many areas of Qtile that I do not personally use. I rely on community reviews to vet pull requests against these areas.

> I hope that either xkb-switch is going to eventually become official 

What does "become official" mean, exactly? Become more widely used? It can only become more widely used if people adopt it :). As I said, I'm open to alternative implementations which use more mainstream tools.

> this new implementation brings a ton of value with it.

It fixed a bug someone reported, and sat on the PR tracker for two months with no negative reviews. In my mind, that says it adds value. If you don't like it, please open a new issue with your reasoning why, and we can discuss it there.

In general, I appreciate the feedback, but it is more useful *before* stuff lands :)

**Comment by tych0** [comment]
I can appreciate the "this is an obscure dependency" argument, so thanks for raising this issue. If you have this complaint, I'm sure others do too. I filed #2265 for further discussion.

**Pull request merged**